### PR TITLE
feat: Support rendering 'Oblique' fonts

### LINKED ISSUE #235: Rendering 'italic' or 'oblique' text misses fonts with 'Oblique' subfamily
When `TextRenderer.prepareRenderParams()` sees a `fontStyle` of `italic` or `oblique` the FontRegistry only looks for a TrueTypeFontFile where `isItalic()` is True, i.e. whre the subfamily is 'Italic'.

So cursive fonts like DejaVuSans-Oblique.ttf or DejaVuSans-BoldOblique.ttf are not detected and the text is rendered 'normal'.

The spec says, that 'italic' should fall back to an 'oblique' font and 'oblique' should fall back to an 'italic' one.
Not talking about creating an artficially cursive simulation, 

I think a `TrueTypeFontFile->isOblique()` function should be introduced and `FontRegistry.findMatchingFont()` should look for the fallback. Of course, its `bool $italic` parameter has to be adapted accordingly.

### PR DESCRIPTION
**Pull request by Tontyna**
resolves  #235

### CODE REVIEW

**Review by meyfa — COMMENTED** [review]
Changes look good to me, thanks @Tontyna! Could you fix the indentation to make PHPCS happy?

**Comment by Tontyna** [comment]
of course, if only I knew the rules...  Tried to guess.

File: src/Fonts/FontRegistry.php [hunk 1]
```diff
@@ -32,9 +32,19 @@ public function findMatchingFont(?string $family, bool $italic, float $weight):
...
 
-        // Attempt to find the closest-weight match with correct family and italicness.
-        $match = $this->closestMatchBasedOnWeight(function (FontFile $font) use ($family, $anyFontFamily, $italic) {
-            return ($anyFontFamily || $font->getFamily() === $family) && $font->isItalic() === $italic;
+        // Attempt to find the closest-weight match with correct family and cursiveness.
+        $match = $this->closestMatchBasedOnWeight(function (FontFile $font) use ($family, $anyFontFamily, $style) {
+            $result = ($anyFontFamily || $font->getFamily() === $family); 
+            $isItalic = $font->isItalic();
+            $isOblique = $font->isOblique();
+            switch ($style){
```
Comment (meyfa) [inline src/Fonts/FontRegistry.php:40]:
```suggestion
            switch ($style) {
```

File: src/Fonts/FontRegistry.php [hunk 2]
```diff
@@ -32,9 +32,19 @@ public function findMatchingFont(?string $family, bool $italic, float $weight):
             }
         }
 
-        // Attempt to find the closest-weight match with correct family and italicness.
-        $match = $this->closestMatchBasedOnWeight(function (FontFile $font) use ($family, $anyFontFamily, $italic) {
-            return ($anyFontFamily || $font->getFamily() === $family) && $font->isItalic() === $italic;
+        // Attempt to find the closest-weight match with correct family and cursiveness.
+        $match = $this->closestMatchBasedOnWeight(function (FontFile $font) use ($family, $anyFontFamily, $style) {
+            $result = ($anyFontFamily || $font->getFamily() === $family);
```
Comment (meyfa) [inline src/Fonts/FontRegistry.php:37]: There is a trailing space:

```suggestion
            $result = ($anyFontFamily || $font->getFamily() === $family);
```

File: src/Fonts/TrueTypeFontFile.php [hunk 3]
```diff
@@ -44,14 +44,23 @@ public function getFamily(): string
 
     public function getWeight(): float
     {
-        return $this->weightClass ?? ($this->subfamily === 'Bold' || $this->subfamily === 'Bold Italic' ? 700 : 400);
+        return $this->weightClass ?? (
+            $this->subfamily === 'Bold' ||
```
Comment (meyfa) [inline src/Fonts/TrueTypeFontFile.php:48]: Also one here:

```suggestion
            $this->subfamily === 'Bold' ||
```

**Review by meyfa — CHANGES_REQUESTED** [review]

**Review by meyfa — APPROVED** [review]

**Comment by meyfa** [comment]
Thank you again for reporting this bug & also fixing it! This is now released as [v0.16.0](https://github.com/meyfa/php-svg/releases/tag/v0.16.0).

**Pull request merged**